app/ui-react/syndesis/src/modules/data/ViewsImportApp.tsx

```diff
@@ -18,6 +18,43 @@ export const ViewsImportApp: React.FunctionComponent = () => {
   const { t } = useTranslation(['data', 'shared']);
   const { state } = useRouteData<null, IViewsImportAppRouteState>();
 
+  const [selectedConnection, setSelectedConnection] = React.useState('');
+
+  const handleConnectionSelectionChanged = async (
+    name: string,
+    selected: boolean
+  ) => {
+    const selConn = selected ? name : '';
+    setSelectedConnection(selConn);
+    clearViewSelection();
+  };
+
+  const [selectedViews, setSelectedViews] = React.useState<ViewInfo[]>([]);
+  const [hasSelectedViews, setHasSelectedViews] = React.useState(false);
+
+  const handleAddView = async (view: ViewInfo) => {
+    const currentViews = selectedViews;
+    currentViews.push(view);
+    setSelectedViews(currentViews);
+    setHasSelectedViews(currentViews.length > 0);
```

Comment:
Couldn't you just do this: `setHasSelectedViews(true);` Also was wondering what you thought about this: Since `setHasSelectedViews` is *always* called after `setSelectedViews` is called, we could move the `setHasSelectedViews` call into a `useEffect` that runs whenever `selectedViews` changes.

Reply:
Yes, Dan we could have used `setHasSelectedViews(true);` here. I may have missed this because I just lifted the state management(code) from `SelectViewsPage` to `ViewsImportApp`.
And while checking your second suggestion about using `useEffect` I found that we are incorrectly updating the React State in the case state being an array. If handled correctly we may not even require `hasSelectedViews` state. I have done some changes locally to verify this. Maybe we can discuss it before I push these changes.